.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      # Prevent committing directly to master
+      - id: no-commit-to-branch
+        args:
+          - --branch=master
+      # Sorts entries in requirements*.txt
+      - id: requirements-txt-fixer
+  - repo: local
+    hooks:
+      - id: mypy
+        name: mypy
+        entry: ./scripts/run-pre-commit.sh mypy
+        language: script
+        types: [python]
+        require_serial: true
+        files: ^(arelle)/.+\.py$
```

Comment:
Briefly experimenting with this, it doesn't appear to respect the [pyproject.toml config](https://github.com/Arelle/Arelle/blob/451b9484d7453f71bb6b20935aa3365125bc1248/pyproject.toml#L24-L31). It's throwing errors on other files.

Reply:
We might have to run it with `--namespace-packages` in the same way we do in tox. There are some folders that don't contain an `__init__.py`, hence this extra argument to mypy.

If that doesn't work. it might be that we have to pass `--config-file pyproject.toml` as well.

Reply:
Yes, that did the trick, we need to run it with `--namespace-packages`.

This is another way if testing it before/after adding the option to mypy:
pre-commit run --files arelle/plugin/validate/ESEF/Util.py

Reply:
Experimenting with this more and unfortunately I'm not sure mypy belongs as a pre-commit hook. If I make edits to a reasonable number of python files (10) it takes 10-20 seconds to run mypy on those changes with this hook. That's probably better left to running alongside the test suite.

Reply:
Yeah, mypy runs on the entire codebase in the suggested setup. Should we drop the pre-commit hook altogether for now then?

Reply:
Yeah, I think so. Sadly it adds too much overhead to the commit workflow.

Reply:
OK, thanks for considering it though!